### PR TITLE
Endoc 318

### PR DESCRIPTION
**Pull request by jyunmitch**
changed three files

### CODE REVIEW

File: vuepress/docs/next/docs/component-generator/component-gen-tech.md [hunk 1]
```diff
@@ -22,10 +22,8 @@ When generating a microservice using the Entando Component Generator there are a
   * There are many powerful technologies in the marketplace but it is up to the developer to integrate any choices from the marketplace into their Entando application. Entando doesn't provide any direct support for marketplace tech choices.
 * Micro frontends
   * Users can optionally choose to generate micro frontends for entities generated via the blueprint.
-  * The Entando Component Generator only provides micro frontends in React from the generation flow but developers can create and bundle micro frontends in other technologies as well.
+  * The Entando Component Generator only provides micro frontends in React from the generation flow but developers can create and bundle micro frontends in other technologies as well.
```
Comment (jyunmitch) [inline vuepress/docs/next/docs/component-generator/component-gen-tech.md:25]: should we add text about healthCheckPath in this document

Comment (nshaw) [reply vuepress/docs/next/docs/component-generator/component-gen-tech.md:25]: Yep, maybe an extra bullet under "Microservice code" at the top. I would point to the other page for the details.

Comment (Lyd1aCla1r3) [inline vuepress/docs/next/docs/component-generator/component-gen-tech.md:25]: would reorder and add comma; "The ECG generation flow only provides micro frontends in React, but developers..."

File: vuepress/docs/next/docs/component-generator/blueprint-features.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
+# Entando Blueprint Features
+
+The Entando Blueprint makes it easier and faster to customize your application by   generating controllers, repositories, services, and micro frontends for your entity. It means generating those project files in minutes simply by answering questions that define the parameters of your applications.
```
Comment (Lyd1aCla1r3) [inline vuepress/docs/next/docs/component-generator/blueprint-features.md:3]: extra space?
entity sounds odd... i think you can stop the sentence at "frontends"; application is implied
add comma: "in minutes, simply"
applications -> application (singular to be consistent)

File: vuepress/docs/next/docs/component-generator/blueprint-features.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
+# Entando Blueprint Features
+
+The Entando Blueprint makes it easier and faster to customize your application by   generating controllers, repositories, services, and micro frontends for your entity. It means generating those project files in minutes simply by answering questions that define the parameters of your applications. 
+
+The Entando Blueprint uses JHipster Technology based on Embedded JavaScript (EJS), a templating language that provides powerful constructs for large scale file generation. 
+
+## The Features:
+* Backend with Spring Boot 
+  * profiles (dev, prod)
```
Comment (Lyd1aCla1r3) [inline vuepress/docs/next/docs/component-generator/blueprint-features.md:9]: Profiles

File: vuepress/docs/next/docs/component-generator/blueprint-features.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
+# Entando Blueprint Features
+
+The Entando Blueprint makes it easier and faster to customize your application by   generating controllers, repositories, services, and micro frontends for your entity. It means generating those project files in minutes simply by answering questions that define the parameters of your applications. 
+
+The Entando Blueprint uses JHipster Technology based on Embedded JavaScript (EJS), a templating language that provides powerful constructs for large scale file generation.
```
Comment (Lyd1aCla1r3) [inline vuepress/docs/next/docs/component-generator/blueprint-features.md:5]: JHipster Technology is a proper noun?

File: vuepress/docs/next/docs/component-generator/blueprint-features.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
+# Entando Blueprint Features
+
+The Entando Blueprint makes it easier and faster to customize your application by   generating controllers, repositories, services, and micro frontends for your entity. It means generating those project files in minutes simply by answering questions that define the parameters of your applications. 
+
+The Entando Blueprint uses JHipster Technology based on Embedded JavaScript (EJS), a templating language that provides powerful constructs for large scale file generation. 
+
+## The Features:
+* Backend with Spring Boot 
+  * profiles (dev, prod)
+  * Pre-configured Cross-Origin Resource Sharing (CORS) settings
```
Comment (Lyd1aCla1r3) [inline vuepress/docs/next/docs/component-generator/blueprint-features.md:10]: i don't think preconfigured needs a hyphen

File: vuepress/docs/next/docs/component-generator/blueprint-features.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+  * Jhipster Design Language (JDL) support for schema upgrade
+  * Liquibase integration for schema upgrade
+* Frontend 
+  * Localization
+  * Keycloak integration
+  * .env profiles 
+
+For more information:
+* Install [Jhipster Entando Blueprint](https://github.com/entando/generator-jhipster-entando/blob/master/README.md)
+* Try implementing an Entando Blueprint-- [Create micro services and micro Frontents](../../tutorials/micro-frontends/react.html)
```
Comment (Lyd1aCla1r3) [inline vuepress/docs/next/docs/component-generator/blueprint-features.md:23]: microservices

File: vuepress/docs/next/docs/component-generator/blueprint-features.md [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+
+## Premade Widgets
+When you create an entity using the Entando Blueprint, it generates a few
```
Comment (Lyd1aCla1r3) [inline vuepress/docs/next/docs/component-generator/blueprint-features.md:26]: entity sounds odd again... "Several premade widgets are generated when you use the Entando Blueprint to create an application."

File: vuepress/docs/next/docs/component-generator/blueprint-features.md [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+premade widgets. Below, each will be reviewed.
+* [Authentication](#authentication)
+* [Custom events](#custom-events)
+* [Tests and mocks](#tests-and-mocks)
+* [PropTypes](#proptypes)
+* [Fetching data](#fetching-data)
+* [Form widget](#form-widget)
+
+We will be using `Conference` as an entity name for the example below.
```
Comment (Lyd1aCla1r3) [inline vuepress/docs/next/docs/component-generator/blueprint-features.md:35]: what kind of an entity is 'Conference'?

File: vuepress/docs/next/docs/component-generator/blueprint-features.md [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+
+What the widgets have in **common**:
+* Each widget contains a README file that helps with the setup.
+
+* All generated widgets are web components created using Custom Elements
```
Comment (Lyd1aCla1r3) [inline vuepress/docs/next/docs/component-generator/blueprint-features.md:40]: "All widgets are web components created with the Custom Elements API."

File: vuepress/docs/next/docs/component-generator/blueprint-features.md [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+API.
+
+* Each widget is displayed using the custom element tag - e.g., inside the
```
Comment (Lyd1aCla1r3) [inline vuepress/docs/next/docs/component-generator/blueprint-features.md:43]: "...tag. For example, inside..."

File: vuepress/docs/next/docs/component-generator/blueprint-features.md [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+details widget folder `conference/detailsWidget/public/index.html`, you
+can find
+`conference-details id="1" override-edit-handler hide-edit-button />`.
+This element `<conference-details />` is defined in the component entry
+point at
+`conference/detailsWidget/src/custom-elements/ConferenceDetailsElement.js`.
+
+> **Note**
+>
+> custom element names (`conference-details`) require a dash in them to
```
Comment (Lyd1aCla1r3) [inline vuepress/docs/next/docs/component-generator/blueprint-features.md:53]: Custom
would reword... "The custom element naming convention requires a hyphen, e.g. `conference-details`. Single words are not valid custom element names."

File: vuepress/docs/next/docs/component-generator/blueprint-features.md [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+      </KeycloakContext.Provider>,
+      this.mountPoint
+    );
+
+For more information about the authentication process, please refer
+to the [Authentication section](../reference/identity-management.html#authentication).
+
+### Custom events
+
+All widgets rely on custom events for communication which is why custom
```
Comment (Lyd1aCla1r3) [inline vuepress/docs/next/docs/component-generator/blueprint-features.md:81]: would reword... "All widgets rely on custom events for communication and contain event creation and removal, as well as event listener creation."

File: vuepress/docs/next/docs/component-generator/blueprint-features.md [hunk 13]
```diff
@@ -0,0 +1,135 @@
...
+event creation and removal is in each widget, as well as event listener
+creation. Note that when an event listener is created, it should be when
```
Comment (Lyd1aCla1r3) [inline vuepress/docs/next/docs/component-generator/blueprint-features.md:83]: "Note that an event listener should be created when the element is no longer needed."

File: vuepress/docs/next/docs/component-generator/blueprint-features.md [hunk 14]
```diff
@@ -0,0 +1,135 @@
...
+the element is no longer needed. It should be created in the custom
+element’s `disconnectedCallback()` function.
+
+To add more events to listen to the widgets, add the event types to
```
Comment (Lyd1aCla1r3) [inline vuepress/docs/next/docs/component-generator/blueprint-features.md:87]: would reword: "To add more event listeners, add the event types to the `INPUT_EVENT_TYPES` object at `detailsWidget/src/custom-elements/widgetEventTypes.js`."

File: vuepress/docs/next/docs/component-generator/blueprint-features.md [hunk 15]
```diff
@@ -0,0 +1,135 @@
...
+`INPUT_EVENT_TYPES` object at
+`detailsWidget/src/custom-elements/widgetEventTypes.js` which will add
+it to the listener list. To remove the event, simply remove the element from the list.
```
Comment (Lyd1aCla1r3) [inline vuepress/docs/next/docs/component-generator/blueprint-features.md:90]: an event

File: vuepress/docs/next/docs/component-generator/blueprint-features.md [hunk 16]
```diff
@@ -0,0 +1,135 @@
...
+
+For more information about custom events and widget communication,
+please refer to the section on [Widget communication](../../tutorials/micro-frontends/communication.html).
+
+### Tests and mocks
+
+Each widget has tests written for it. Entando uses
+`react-testing-library`, but developers are free to upgrade and use any
+tool desired. Tests are kept at `detailsWidget/src/components/` and mocks for
```
Comment (Lyd1aCla1r3) [inline vuepress/docs/next/docs/component-generator/blueprint-features.md:99]: "Tests and their mocks are kept in `detailsWidget/src/components/`."

File: vuepress/docs/next/docs/component-generator/blueprint-features.md [hunk 17]
```diff
@@ -0,0 +1,135 @@
...
+point at
+`conference/detailsWidget/src/custom-elements/ConferenceDetailsElement.js`.
+
+> **Note**
+>
+> custom element names (`conference-details`) require a dash in them to
+> be used, (kebab-case) - they can not be single words.
+
+For more information about web components, custom elements and micro
+frontends, refer to [Create a React micro frontend widget](../../tutorials/micro-frontends/react.html)
```
Comment (Lyd1aCla1r3) [inline vuepress/docs/next/docs/component-generator/blueprint-features.md:57]: missing period

File: vuepress/docs/next/docs/component-generator/blueprint-features.md [hunk 18]
```diff
@@ -0,0 +1,135 @@
...
+Each widget has tests written for it. Entando uses
+`react-testing-library`, but developers are free to upgrade and use any
+tool desired. Tests are kept at `detailsWidget/src/components/` and mocks for
+them are at `detailsWidget/src/components/`.
+
+### PropTypes
+
+PropTypes for data used across several components are shared. You can
+see and modify them at `detailsWidget/src/components/`. This way you can
+avoid repeating the same propTypes in each component and import the shared
```
Comment (Lyd1aCla1r3) [inline vuepress/docs/next/docs/component-generator/blueprint-features.md:106]: would reword to "by importing"

File: vuepress/docs/next/docs/component-generator/blueprint-features.md [hunk 19]
```diff
@@ -0,0 +1,135 @@
...
+### Fetching data
+
+For data fetching from widgets, use Fetch API. You can find the functions for
+fetching data at `detailsWidget/src/api`, in different files for
+different contexts.
+
+## Form widget
+
+For displaying forms within a widget, use
+[Formik](https://jaredpalmer.com/formik) which helps with form state
```
Comment (Lyd1aCla1r3) [inline vuepress/docs/next/docs/component-generator/blueprint-features.md:132]: missing comma: ".../formik), which..."

File: vuepress/docs/next/docs/component-generator/blueprint-features.md [hunk 20]
```diff
@@ -0,0 +1,135 @@
...
+
+### Fetching data
+
+For data fetching from widgets, use Fetch API. You can find the functions for
+fetching data at `detailsWidget/src/api`, in different files for
+different contexts.
+
+## Form widget
+
+For displaying forms within a widget, use
```
Comment (Lyd1aCla1r3) [inline vuepress/docs/next/docs/component-generator/blueprint-features.md:131]: To display

File: vuepress/docs/next/docs/component-generator/blueprint-features.md [hunk 21]
```diff
@@ -0,0 +1,135 @@
...
+    ConferenceDetails.propTypes = {
+      conference: conferenceType,
+      t: PropTypes.func.isRequired,
+    };
+
+    export default ConferenceDetails;
+
+### Fetching data
+
+For data fetching from widgets, use Fetch API. You can find the functions for
```
Comment (Lyd1aCla1r3) [inline vuepress/docs/next/docs/component-generator/blueprint-features.md:125]: To fetch data from widgets,

File: vuepress/docs/next/docs/component-generator/blueprint-features.md (hunk 4 above)
Comment (jyunmitch) [reply vuepress/docs/next/docs/component-generator/blueprint-features.md:5]: it's not

File: vuepress/docs/next/docs/component-generator/blueprint-features.md (hunk 9 above)
Comment (jyunmitch) [reply vuepress/docs/next/docs/component-generator/blueprint-features.md:40]: we are only talking about the generated widgets

File: vuepress/docs/next/docs/component-generator/blueprint-features.md (hunk 11 above)
Comment (jyunmitch) [reply vuepress/docs/next/docs/component-generator/blueprint-features.md:53]: reworded

File: vuepress/docs/next/docs/component-generator/blueprint-features.md (hunk 13 above)
Comment (jyunmitch) [reply vuepress/docs/next/docs/component-generator/blueprint-features.md:83]: the important thing is **when** the event listener is created in the disconnectedCallback() function, not when the listener is being created in the process.

File: vuepress/docs/next/docs/component-generator/blueprint-features.md (hunk 20 above)
Comment (jyunmitch) [reply vuepress/docs/next/docs/component-generator/blueprint-features.md:131]: I think these are helper apps. One doesn't directly do the other, Formik doesn't display the forms, it helps manage the data.

File: vuepress/docs/next/docs/component-generator/blueprint-features.md [hunk 22]
```diff
@@ -0,0 +1,130 @@
...
+The Entando Blueprint uses JHipster technology based on Embedded JavaScript (EJS), a templating language that provides powerful constructs for large scale file generation. 
+
+## The Features:
+* Backend with Spring Boot 
+  * Profiles (dev, prod)
+  * Preconfigured Cross-Origin Resource Sharing (CORS) settings
+  * Keycloak integration 
+  * Swagger/OpenAPI frontend
+  * Data modeling 
+  * Jhipster Design Language (JDL) support for schema upgrade
```
Comment (nshaw) [inline vuepress/docs/next/docs/component-generator/blueprint-features.md:14]: JHipster (capital H)

File: vuepress/docs/next/docs/component-generator/blueprint-features.md [hunk 23]
```diff
@@ -0,0 +1,130 @@
...
+  * Liquibase integration for schema upgrade
+* Frontend 
+  * Localization
+  * Keycloak integration
+  * .env profiles 
+
+For more information:
+* Install [Jhipster Entando Blueprint](https://github.com/entando/generator-jhipster-entando/blob/master/README.md)
```
Comment (nshaw) [inline vuepress/docs/next/docs/component-generator/blueprint-features.md:22]: JHipster

File: vuepress/docs/next/docs/component-generator/blueprint-features.md (hunk 8 above)
Comment (nshaw) [reply vuepress/docs/next/docs/component-generator/blueprint-features.md:35]: It's our standard example for a simple data model, like attending a conference, so it could have a name, location, date, etc.

File: vuepress/docs/next/docs/component-generator/component-gen-tech.md (hunk 1 above)
Comment (nshaw) [reply vuepress/docs/next/docs/component-generator/component-gen-tech.md:26]: "using other technologies" instead of "in other technologies"

File: vuepress/docs/next/docs/component-generator/blueprint-features.md [hunk 24]
```diff
@@ -0,0 +1,130 @@
...
+## The Features:
+* Backend with Spring Boot 
+  * Profiles (dev, prod)
+  * Preconfigured Cross-Origin Resource Sharing (CORS) settings
+  * Keycloak integration 
+  * Swagger/OpenAPI frontend
+  * Data modeling 
+  * Jhipster Design Language (JDL) support for schema upgrade
+  * Liquibase integration for schema upgrade
+* Frontend
```
Comment (nshaw) [inline vuepress/docs/next/docs/component-generator/blueprint-features.md:16]: Frontend with React

File: vuepress/docs/next/docs/component-generator/blueprint-features.md [hunk 25]
```diff
@@ -0,0 +1,130 @@
+# Entando Blueprint Features
+
+The Entando Blueprint makes it easier and faster to customize your application by generating controllers, repositories, services, and micro frontends for your entity. It means generating those project files in minutes, simply by answering questions that define the parameters of your application. 
+
+The Entando Blueprint uses JHipster technology based on Embedded JavaScript (EJS), a templating language that provides powerful constructs for large scale file generation. 
+
+## The Features:
+* Backend with Spring Boot 
+  * Profiles (dev, prod)
```
Comment (nshaw) [inline vuepress/docs/next/docs/component-generator/blueprint-features.md:9]: Maybe alpha-sort the bullets?

File: vuepress/docs/next/docs/component-generator/blueprint-features.md (hunk 22 above)
Comment (nshaw) [reply vuepress/docs/next/docs/component-generator/blueprint-features.md:14]: This bullet overlaps with the next one. Let's keep this as just "JHipster Design Language (JDL) support" and the  next one as-is. The JDL support is useful even without Liquibase.

File: vuepress/docs/next/docs/component-generator/component-gen-tech.md [hunk 26]
```diff
@@ -6,7 +6,8 @@ When generating a microservice using the Entando Component Generator there are a
 
 * Microservice code
   * The Component Generator currently only supports the generation of Spring Boot microservices but will support the generation of Quarkus based services soon. 
-  * If you're interested in using Quarkus in your Entando app contact us on the [Forum](https://forum.entando.org/) or [Community Slack](https://join.slack.com/t/entandocommunity/shared_invite/zt-g609owdv-2K~YRh8zrI6lqlWo4aFWUw) and we'll help you get started.
+  * If you're interested in using Quarkus in your Entando app, contact us on the [Forum](https://forum.entando.org/) or [Community Slack](https://join.slack.com/t/entandocommunity/shared_invite/zt-g609owdv-2K~YRh8zrI6lqlWo4aFWUw) and we'll help you get started.
+  * If you're considering non-Java based microservices, here is a definition of an Entando plugin, including information on the runtime contract required to use them.
```
Comment (nshaw) [inline vuepress/docs/next/docs/component-generator/component-gen-tech.md:10]: Is there a missing link? Seems like "here" should point somewhere.

**Review by nshaw — CHANGES_REQUESTED** [review]
A few suggestions/fixes in the open conversations

File: vuepress/docs/next/docs/component-generator/component-gen-tech.md (hunk 26 above)
Comment (jyunmitch) [reply vuepress/docs/next/docs/component-generator/component-gen-tech.md:10]: oops. good catch

**Review by nshaw — APPROVED** [review]
